src/zenml/orchestrators/step_run_utils.py

```diff
@@ -0,0 +1,566 @@
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utilities for creating step runs."""
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Dict, Mapping, Optional, Set, Tuple
+
+from zenml.client import Client
+from zenml.config.step_configurations import ArtifactConfiguration, Step
+from zenml.constants import CODE_HASH_PARAMETER_NAME, TEXT_FIELD_MAX_LENGTH
+from zenml.enums import ExecutionStatus
+from zenml.logger import get_logger
+from zenml.model.utils import link_artifact_version_to_model_version
+from zenml.models import (
+    ArtifactVersionResponse,
+    ModelVersionPipelineRunRequest,
+    ModelVersionResponse,
+    PipelineDeploymentResponse,
+    PipelineRunResponse,
+    PipelineRunUpdate,
+    StepRunRequest,
+    StepRunResponse,
+    StepRunUpdate,
+)
+from zenml.orchestrators import cache_utils, input_utils, utils
+from zenml.stack import Stack
+from zenml.utils import string_utils
+
+if TYPE_CHECKING:
+    from zenml.model.model import Model
+
+logger = get_logger(__name__)
+
+
+class StepRunRequestFactory:
+    """Helper class to create step run requests."""
+
+    def __init__(
+        self,
+        deployment: "PipelineDeploymentResponse",
+        pipeline_run: "PipelineRunResponse",
+        stack: "Stack",
+    ) -> None:
+        """Initialize the object.
+
+        Args:
+            deployment: The deployment for which to create step run requests.
+            pipeline_run: The pipeline run for which to create step run
+                requests.
+            stack: The stack on which the pipeline run is happening.
+        """
+        self.deployment = deployment
+        self.pipeline_run = pipeline_run
+        self.stack = stack
+
+    def create_request(self, invocation_id: str) -> StepRunRequest:
+        """Create a step run request.
+
+        This will only create a request with basic information and will not yet
+        compute information like the cache key and inputs. This is separated
+        into a different method `populate_request(...)` that might raise
+        exceptions while trying to compute this information.
+
+        Args:
+            invocation_id: The invocation ID for which to create the request.
+
+        Returns:
+            The step run request.
+        """
+        return StepRunRequest(
+            name=invocation_id,
+            pipeline_run_id=self.pipeline_run.id,
+            deployment=self.deployment.id,
+            status=ExecutionStatus.RUNNING,
```

Comment:
Why not `INITIALIZING`?

Reply:
Somehow we never used that state for steps and only for pipelines. I guess there are only a few cases:
- If something fails, we change the state to failed.
- If everything goes right, the step status changes to running in less than 1 second after this request is made. So we would only need to update the status immediately anyway, which I think we can avoid by just setting it to running from the start?

Reply:
Ok, let's keep it.